### ISSUE TITLE
Multimap with prefilter working unreliably

### ISSUE THREAD

**Issue by Artim96** [rspamd/rspamd#4426]
Hi,
for quite a long time, I'm trying to establish a sender domain based acceptlist/blocklist ruleset in a way that when a mail comes from either list it's only marked as accepted or as spam without any further filtering. So even if the mail had been (falsely) identified as spam, rspamd would ignore that. For all I know that should be possible via the multimap, but it's unreliable at best. Here's what I'm doing, maybe someone sees something I'm getting wrong:

local.d/multimap.conf:
```
ACCEPTLIST_SENDER_DOMAIN {
      #type = "header";
      #header = "from";
      type = "from";
      prefilter = "true";
      filter = "email:domain";
      map = "file:///etc/rspamd/local.d/acceptlist.sender.domain.map";
      description = "Sender domain is on allowed";
      action = "accept"; # Prefilter mode
  }

BLOCKLIST_SENDER_DOMAIN {
      #type = "header";
      #header = "from";
      type = "from";
      prefilter = "true";
      filter = "email:domain";
      map = "file:///etc/rspamd/local.d/blocklist.sender.domain.map";
      symbol = "BLOCKLIST_SENDER_DOMAIN";
      description = "Sender domain is on blocked";
      score = 10.0;
      action = "add_header"; # Prefilter mode
  }
```
The two comments at the top of each block is from when I tried to filter for header and then the from header, but since you can directly filter for that I tried this one, it doesn't help though. Also, I tried adding the files in map with both just the path and `file:///path/to/file`, still no success either way. The rspamd.log also doesn't show anything helpful. Any ideas?

I'm on Debian 11 running rspamd 3.4-1~bullseye straight from the rspamd repo.

_Originally posted by @Artim96 in https://github.com/rspamd/rspamd/discussions/4392_

**Comment by Artim96** [comment]
As nobody is able to tell what's wrong with it and since it does sometimes work, I expect this to be a bug, thus the issue created from the discussion.

**Comment by vstakhov** [comment]
I don't understand your issue. `type=from` and `type=header` are different type of maps: the first matches **SMTP from** while the second uses **MIME from**. It is very common that these addresses are different and use different domains.

**Comment by vstakhov** [comment]
I'm sorry, but what you describe is not likely an Rspamd issue. You'd need to understand how email works but nonetheless the issue tracker is not the appropriate place for such a discussion. Closing issue.

**Comment by Artim96** [comment]
Excuse me? Rspamd is obviously not behaving as intended, at least not reliably. This is exactly the right place to discuss this. Also, I did try in discussions first but got ignored. So do tell me, how should I get your product to behave as you intend if you refuse to give proper explanation on how to configure it? I followed the documentation in two different ways, neither results in a reliable behavior. That means either the documentation is bad or rspamd is simply buggy.
